src/public/videoApp.ts

```diff
@@ -0,0 +1,154 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FrameContexts } from './constants';
+import { registerHandler } from '../internal/handlers';
+/**
+ * Namespace to video extensibility of the SDK.
+ *
+ */
+export namespace videoApp {
+  /**
+   * Represents a video frame.
+   */
+  export interface VideoFrame {
+    /**
+     * Video frame width.
+     */
+    width: number;
+    /**
+     * Video frame height.
+     */
+    height: number;
+    /**
+     * Video frame buffer
+     */
+    data: Uint8ClampedArray;
+    /**
+     * NV12 luma stride, valid only when video frame format is NV12
+     */
+    lumaStride: number;
+    /**
+     * NV12 chroma stride, valid only when video frame format is NV12
+     */
+    chromaStride: number;
+    /**
+     * RGB stride, valid only when video frame format is RGB
+     */
+    stride: number;
+  }
+  /**
+   * Video frame format enum, currentlyl only support NV12
+   */
+  export enum VideoFrameFormat {
+    NV12,
+  }
+  /**
+   * Video frame configuration supplied to Teams to customize the generated video frame parameters, like format.
+   */
+  export interface VideoFrameConfig {
+    /**
+     * video format
+     */
+    format: VideoFrameFormat;
+  }
+
+  /**
+   *  Video effect change type enum
+   */
+  export enum EffectChangeType {
+    /**
+     * current video effect changed.
+     */
+    EffectChanged,
+    /**
+     * disable the video effect
+     */
+    EffectDisabled,
+  }
+
+  /**
+   *  Video frame call back funtion definition
+   */
+  type VideoFrameCallback = (
+    frame: VideoFrame,
+    notifyVideoFrameProcessed: () => void,
+    notifyError: (errorMessage: string) => void,
+  ) => void;
+
+  /**
+   *  Video effect change call back funtion definition
+   */
+  type VideoEffectCallBack = (effectName: string | undefined) => void;
+
+  /**
+   * VideoApp
+   */
+  class VideoApp {
```

Comment:
we follow namespace and all others functions are stateless.

Reply:
Please update to same format.

Reply:
Thanks for pointing it out.